### PR TITLE
fix generics in RecordDiffer.java

### PR DESCRIPTION
**Pull request by stephane-airbyte**
[RecordDiffer.buildIdentityComparator](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java#L155) is causing me troubles when converting to kotlin.
I think the reason is the use of raw types, which I tracked down to the [extract](https://github.com/airbytehq/airbyte/blob/a8fa18ff3b5377dfd39598283eb8262e09dec095/airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java#L436)

I think I'm doing what was initially intended, but I'd really appreciate several extra pairs of eyes on this!

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
* **#36462** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36462?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java [hunk 1]
```diff
@@ -432,10 +433,28 @@ private static LocalDate asDate(final JsonNode node) {
     }
   }
 
+  private static class Field implements Comparable<Field> {
+    String stringValue;
```
Comment (jdpgrailsdev) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java:437]: Nit:  can these fields be private?  Does the wrapping class need access to them?

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Review by edgao — APPROVED** [review]
seems right to me! Manually tested locally by breaking a test in destination-snowflake + verified correct assertion failures:
image

(at some point I should really write some unit tests for this class >.>)

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java [hunk 2]
```diff
@@ -432,10 +433,30 @@ private static LocalDate asDate(final JsonNode node) {
...
+    public int compareTo(@NotNull RecordDiffer.Field o) {
+      if (realType.getCanonicalName().equals(o.realType.getCanonicalName())) {
+        return stringValue.compareTo(o.stringValue);
+      }
+      return realType.getCanonicalName().compareTo(o.realType.getCanonicalName());
+    }
+
+  }
+
   // Generics? Never heard of 'em. (I'm sorry)
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/RecordDiffer.java:456]: you could probably kill this comment now :P

**Review by gisripa — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Mar 25, 6:37 PM EDT**: @stephane-airbyte merged this pull request with [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/36462).

**Pull request merged**